### ISSUE TITLE
Amaze stops/crashes on opening files with external programs

### ISSUE THREAD

**Issue by madduck** [TeamAmaze/AmazeFileManager#1568]
Any time I try to open a file with an external viewer (e.g. a PDF with mupdf, or a JPEG with an image viewer), the external app is correctly launched, but then Android tells me that "Amaze has stopped." and gives me the option to click "Close app".

This is using the latest version from F-Droid, v3.3.2.

![Screenshot showing the "close app" dialog](https://user-images.githubusercontent.com/195073/51092933-5a918c00-1802-11e9-9e6b-140fc6865a9e.jpg)

**Comment by EmmanuelMess** [comment]
Do you have a logcat dump?

**Comment by madduck** [comment]
Hey @EmmanuelMess, I am sorry for forgetting about this. Here is a logcat filtered for `/amaze|penandpdf/' as I am opening a PDF with the "Pen&PDF" app:

```
07-30 21:50:26.019  4044 32349 I PackageManager: Setting last chosen activity com.cgogolin.penandpdf/.PenAndPDFActivity for user 0:
07-30 21:50:26.338  4044 32349 I ActivityManager: START u0 {act=android.intent.action.VIEW typ=application/pdf flg=0x13080000 cmp=ComponentInfo{com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity}} from uid 10286
07-30 21:50:26.344  4044  5245 D ActivityManagerPerformance: Received MSG_CFMS_HINT_AMS_SWITCH pkgName: com.cgogolin.penandpdf
07-30 21:50:26.368  4044  4149 D ViewRootImpl@d986fae[penandpdf]: setView = DecorView@dc4e5e5[penandpdf] TM=true MM=false
07-30 21:50:26.369  4044  4149 D ViewRootImpl@d986fae[penandpdf]: dispatchAttachedToWindow
07-30 21:50:26.369  4044  4149 V WindowManager: Relayout Window{a82c14f u0 Splash Screen com.cgogolin.penandpdf}: viewVisibility=0 req=1439x2475 WM.LayoutParams{(0,0)(fillxfill) sim=#20 ty=3 fl=#81830118 pfl=0x20011 wanim=0x10302fd vsysui=0x600 needsMenuKey=2 colorMode=0 naviIconColor=0}
07-30 21:50:26.372  3630  3630 I SurfaceFlinger: id=7152 createSurf (1440x2560),1 flag=404, Splash Screen com.cgogolin.penandpdf#0
07-30 21:50:26.376  4044  4149 D ViewRootImpl@d986fae[penandpdf]: Relayout returned: old=[0,0][0,0] new=[0,0][1440,2560] result=0x7 surface={valid=true 516947898368} changed=true
07-30 21:50:26.393  4044  4314 D GameManagerService: handleForegroundChange(). pkgName: com.cgogolin.penandpdf, clsName: com.cgogolin.penandpdf.PenAndPDFActivity,FgActivityName:com.cgogolin.penandpdf/.PenAndPDFActivity
07-30 21:50:26.393  4044  4314 D GameManagerService: notifyResumePause(). pkg: com.cgogolin.penandpdf, type: 4, isMinimized: false, isTunableApp: false
07-30 21:50:26.393  4044 12884 D MARsPolicyManager: onPackageResumedFG pkgName = com.cgogolin.penandpdf, userId = 0
07-30 21:50:26.401  4044  4149 V WindowManager: finishDrawingLocked: mDrawState=COMMIT_DRAW_PENDING Window{a82c14f u0 Splash Screen com.cgogolin.penandpdf} in Surface(name=Splash Screen com.cgogolin.penandpdf)
07-30 21:50:26.467  4044  4149 D ViewRootImpl@d986fae[penandpdf]: MSG_RESIZED: frame=Rect(0, 0 - 1440, 2560) ci=Rect(0, 84 - 0, 0) vi=Rect(0, 84 - 0, 0) or=1
07-30 21:50:26.467  4044  5462 V WindowManager: Relayout Window{131124a u0 com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity}: viewVisibility=8 req=1440x2560 WM.LayoutParams{(0,0)(fillxfill) sim=#110 ty=1 fl=#85810100 pfl=0x20000 wanim=0x10302fd vsysui=0x700 needsMenuKey=2 colorMode=0 naviIconColor=0}
07-30 21:50:26.473  4044  6897 D GamePkgDataHelper: getGamePkgDataIncServer(). com.cgogolin.penandpdf
07-30 21:50:26.474  4044  6897 D GameManagerService: identifyGamePackage. com.cgogolin.penandpdf
07-30 21:50:26.474  4044  6897 D GameManagerService: identifyGamePackage. com.cgogolin.penandpdf
07-30 21:50:26.475  4044  6897 D GameManagerService: identifyGamePackage. com.cgogolin.penandpdf
07-30 21:50:26.477  4044  6897 D GameManagerService: identifyGamePackage. com.cgogolin.penandpdf
07-30 21:50:26.479  4044  6897 D GameManagerService: identifyGamePackage. com.cgogolin.penandpdf
07-30 21:50:26.483  4044  4149 V WindowManager: Relayout Window{a82c14f u0 Splash Screen com.cgogolin.penandpdf}: viewVisibility=0 req=1440x2560 WM.LayoutParams{(0,0)(fillxfill) sim=#20 ty=3 fl=#81830118 pfl=0x20011 wanim=0x10302fd vsysui=0x600 needsMenuKey=2 colorMode=0 naviIconColor=0}
07-30 21:50:26.485  4044  4149 D ViewRootImpl@d986fae[penandpdf]: Relayout returned: old=[0,0][1440,2560] new=[0,0][1440,2560] result=0x1 surface={valid=true 516947898368} changed=false
07-30 21:50:26.505  3630  3630 D SurfaceFlinger:       GLES | 6fd442de40 | 0000 | 0020 | 00 | 0105 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity#0
07-30 21:50:26.505  3630  3630 D SurfaceFlinger: 0,  542, 1440, 2560 | Splash Screen com.cgogolin.penandpdf#0
07-30 21:50:26.522  3630  3630 D SurfaceFlinger:       GLES | 6fd442de40 | 0000 | 0020 | 00 | 0105 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity#0
07-30 21:50:26.522  3630  3630 D SurfaceFlinger: 0,  452, 1440, 2560 | Splash Screen com.cgogolin.penandpdf#0
07-30 21:50:26.537  4044  5440 V WindowManager: Relayout Window{131124a u0 com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity EXITING}: viewVisibility=8 req=1440x2560 WM.LayoutParams{(0,0)(fillxfill) sim=#20 ty=1 fl=#85810100 pfl=0x20000 wanim=0x10302fd vsysui=0x700 needsMenuKey=2 colorMode=0 naviIconColor=0}
07-30 21:50:26.576 12561 12561 D ViewRootImpl@c3e164f[PenAndPDFActivity]: setView = DecorView@155c1dc[PenAndPDFActivity] TM=true MM=false
07-30 21:50:26.579 12561 12561 W Activity: AppLock checkAppLockState locked:false verifying:false pkgName = com.cgogolin.penandpdf isInMultiWindowMode:false showWhenLocked:false
07-30 21:50:26.581 12561 12561 D ViewRootImpl@c3e164f[PenAndPDFActivity]: dispatchAttachedToWindow
07-30 21:50:26.584  4044  5440 V WindowManager: Relayout Window{f3c9c3f u0 com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity}: viewVisibility=0 req=1439x2475 WM.LayoutParams{(0,0)(fillxfill) sim=#120 ty=1 fl=#81810180 pfl=0x20000 wanim=0x10302fd vsysui=0x600 needsMenuKey=2 colorMode=0 naviIconColor=0}
07-30 21:50:26.589  3630  3630 I SurfaceFlinger: id=7153 createSurf (1440x2560),1 flag=404, com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity#0
07-30 21:50:26.593 13306 13306 E AndroidRuntime: Process: com.amaze.filemanager, PID: 13306
07-30 21:50:26.600  4044 32349 W ActivityManager: crash : com.amaze.filemanager,0
07-30 21:50:26.600 12561 12561 D ViewRootImpl@c3e164f[PenAndPDFActivity]: Relayout returned: old=[0,0][0,0] new=[0,0][1440,2560] result=0x7 surface={valid=true 3391055872} changed=true
07-30 21:50:26.607  4044  4071 I ActivityManager: Showing crash dialog for package com.amaze.filemanager u0
07-30 21:50:26.623  4044  4194 D MdnieScenarioControlService:  packageName : com.cgogolin.penandpdf    className : com.cgogolin.penandpdf.PenAndPDFActivity
07-30 21:50:26.658 12956 12967 W PkgUtils: p: com.amaze.filemanager, u:0
07-30 21:50:26.660 12561 12561 D ViewRootImpl@c3e164f[PenAndPDFActivity]: MSG_RESIZED_REPORT: frame=Rect(0, 0 - 1440, 2560) ci=Rect(0, 84 - 0, 0) vi=Rect(0, 84 - 0, 0) or=1
07-30 21:50:26.660 12561 12561 D ViewRootImpl@c3e164f[PenAndPDFActivity]: MSG_WINDOW_FOCUS_CHANGED 1
07-30 21:50:26.662  4044  4071 V WindowManager: Relayout Window{419cf7a u0 Application Error: com.amaze.filemanager}: viewVisibility=0 req=1352x498 WM.LayoutParams{(0,0)(wrapxwrap) gr=#11 sim=#120 ty=2003 fl=#1820002 pfl=0x110 fmt=-3 wanim=0x10302ec surfaceInsets=Rect(7, 7 - 7, 7) needsMenuKey=2 colorMode=0 naviIconColor=0}
07-30 21:50:26.663 12561 12561 V InputMethodManager: Starting input: tba=android.view.inputmethod.EditorInfo@88caca0 nm : com.cgogolin.penandpdf ic=null
07-30 21:50:26.664  3630  3630 I SurfaceFlinger: id=7154 createSurf (1366x512),1 flag=4, Application Error: com.amaze.filemanager#0
07-30 21:50:26.683  3630  3630 D SurfaceFlinger:        HWC | 6fd442e080 | 0000 | 0020 | 00 | 0100 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | Splash Screen com.cgogolin.penandpdf#0
07-30 21:50:26.700  4044  4149 I WindowManager: Destroying surface Surface(name=com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity) called by com.android.server.wm.WindowStateAnimator.destroySurface:11 com.android.server.wm.WindowStateAnimator.destroySurfaceLocked:216 com.android.server.wm.WindowState.destroyOrSaveSurfaceUnchecked:97 com.android.server.wm.WindowState.destroySurface:137 com.android.server.wm.AppWindowToken.destroySurfaces:28 com.android.server.wm.AppWindowToken.destroySurfaces:1 com.android.server.wm.WindowState.onExitAnimationDone:272 com.android.server.wm.WindowStateAnimator.stepAnimationLocked:553 
07-30 21:50:26.700  3630  3930 I SurfaceFlinger: id=7149 Removed com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity#0 (2/7)
07-30 21:50:26.702  4044  5462 V WindowManager: finishDrawingLocked: mDrawState=COMMIT_DRAW_PENDING Window{f3c9c3f u0 com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity} in Surface(name=com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity)
07-30 21:50:26.702  3630  3706 I SurfaceFlinger: id=7149 Removed com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity#0 (-2/7)
07-30 21:50:26.709  4044  4149 D ViewRootImpl@d986fae[penandpdf]: dispatchDetachedFromWindow
07-30 21:50:26.709  4044  4149 D InputEventReceiver: channel 'a82c14f Splash Screen com.cgogolin.penandpdf (client)' ~ Disposing input event receiver.
07-30 21:50:26.709  4044  4149 D InputEventReceiver: channel 'a82c14f Splash Screen com.cgogolin.penandpdf (client)' ~NativeInputEventReceiver.
07-30 21:50:26.714  3630  3630 I Layer   : id=7149 onRemoved com.amaze.filemanager/com.amaze.filemanager.activities.MainActivity#0 
07-30 21:50:26.720  4044  4079 I ActivityManager: Displayed com.cgogolin.penandpdf/.PenAndPDFActivity: +330ms
07-30 21:50:26.722  4044  4071 V WindowManager: finishDrawingLocked: mDrawState=COMMIT_DRAW_PENDING Window{419cf7a u0 Application Error: com.amaze.filemanager} in Surface(name=Application Error: com.amaze.filemanager)
07-30 21:50:26.740 12561 12561 D ViewRootImpl@c3e164f[PenAndPDFActivity]: MSG_WINDOW_FOCUS_CHANGED 0
07-30 21:50:26.753  3630  3630 D SurfaceFlinger:        HWC | 6fcbe87b00 | 0000 | 0020 | 00 | 0100 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity#0
07-30 21:50:26.753  3630  3630 D SurfaceFlinger:        HWC | 6fd442e2c0 | 0000 | 0020 | 00 | 0105 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | Splash Screen com.cgogolin.penandpdf#0
07-30 21:50:26.769  3630  3630 D SurfaceFlinger:        HWC | 6fd442d780 | 0000 | 0020 | 00 | 0100 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity#0
07-30 21:50:26.769  3630  3630 D SurfaceFlinger:        HWC | 6fd442e2c0 | 0000 | 0020 | 00 | 0105 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | Splash Screen com.cgogolin.penandpdf#0
07-30 21:50:26.769  3630  3630 D SurfaceFlinger:        HWC | 6fcbe86840 | 0000 | 0000 | 00 | 0105 | RGBA_8888   |    0.0,    0.0, 1366.0,  512.0 |   91, 1086, 1348, 1557 | Application Error: com.amaze.filemanager#0
07-30 21:50:26.786  3630  3630 D SurfaceFlinger:        HWC | 6fd442e440 | 0000 | 0020 | 00 | 0100 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity#0
07-30 21:50:26.786  3630  3630 D SurfaceFlinger:        HWC | 6fd442e2c0 | 0000 | 0020 | 00 | 0105 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | Splash Screen com.cgogolin.penandpdf#0
07-30 21:50:26.787  3630  3630 D SurfaceFlinger: 63 | Application Error: com.amaze.filemanager#0
07-30 21:50:26.871  4044  4194 D MdnieScenarioControlService:  packageName : com.cgogolin.penandpdf    className : com.cgogolin.penandpdf.PenAndPDFActivity
07-30 21:50:26.883  4044  4149 I WindowManager: Destroying surface Surface(name=Splash Screen com.cgogolin.penandpdf) called by com.android.server.wm.WindowStateAnimator.destroySurface:11 com.android.server.wm.WindowStateAnimator.destroySurfaceLocked:216 com.android.server.wm.WindowState.destroyOrSaveSurfaceUnchecked:97 com.android.server.wm.WindowState.destroySurface:137 com.android.server.wm.AppWindowToken.destroySurfaces:28 com.android.server.wm.AppWindowToken.destroySurfaces:1 com.android.server.wm.WindowState.onExitAnimationDone:272 com.android.server.wm.WindowStateAnimator.stepAnimationLocked:553 
07-30 21:50:26.883  3630  6756 I SurfaceFlinger: id=7152 Removed Splash Screen com.cgogolin.penandpdf#0 (3/6)
07-30 21:50:26.885  3630  3630 D SurfaceFlinger:        HWC | 6fd442e440 | 0000 | 0020 | 00 | 0100 | RGBA_8888   |    0.0,    0.0, 1440.0, 2560.0 |    0,    0, 1440, 2560 | com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity#0
07-30 21:50:26.885  3630  3630 D SurfaceFlinger:        HWC | 6fcbe86840 | 0000 | 0000 | 00 | 0105 | RGBA_8888   |    0.0,    0.0, 1366.0,  512.0 |   31, 1064, 1409, 1581 | Application Error: com.amaze.filemanager#0
07-30 21:50:26.898  3630  3630 I Layer   : id=7152 onRemoved Splash Screen com.cgogolin.penandpdf#0 
07-30 21:50:27.788 12561 13431 D skia    : Encode PNG Singlethread processname=com.cgogolin.penandpdf :    1089566 us, width=1440, height=720
07-30 21:50:27.895  4044 12884 V WindowManager: Relayout Window{f3c9c3f u0 com.cgogolin.penandpdf/com.cgogolin.penandpdf.PenAndPDFActivity}: viewVisibility=0 req=1440x2560 WM.LayoutParams{(0,0)(fillxfill) sim=#20 ty=1 fl=#81810180 pfl=0x20000 wanim=0x10302fd vsysui=0x600 needsMenuKey=2 colorMode=0 naviIconColor=0}
07-30 21:50:27.902 12561 12561 D ViewRootImpl@c3e164f[PenAndPDFActivity]: Relayout returned: old=[0,0][1440,2560] new=[0,0][1440,2560] result=0x1 surface={valid=true 3391055872} changed=false
```

I cannot discern anything special in there, except of course the application error.

**Comment by VishalNehra** [comment]
Should be fixed with v3.5.0
